use-shopping-cart/core/Entry.js

```diff
@@ -0,0 +1,199 @@
+import { formatCurrencyString } from './store'
+import { v4 as uuidv4 } from 'uuid'
+
+function Entry({
+  product,
+  quantity,
+  currency,
+  language,
+  price_metadata,
+  product_metadata
+}) {
+  quantity = parseInt(quantity)
```

Comment:
We probably shouldn't be using `parseInt` in our library code. I believe you added this when you were working with `<input type="number">`, or perhaps a `<select>`. The number input actually has a built-in way to get the value as a number via `input.valueAsNumber` which is supported in all browsers according to MDN.

Furthermore, `parseInt` takes two arguments otherwise the end result could be messy. It takes `parseInt(string, radix)` where `radix` determines what base you're using. If you don't specify the radix, `parseInt` will guess it based on the string and you could easily end up with base 16 or base 2.

I would suggest leaving the conversion from a string to an integer to the end developers. That being said, we should definitely warn when `count` is not a number **inside the reducers themselves** and then return.

```js
// inside a reducer that takes a count
if (typeof count !== 'number') {
  console.warn('<actionName> requires that count be a number', action)
  return
}
```

Reply:
I made a middleware for throwing warnings. I just didn't want to do it in the reducers themselves, lemme know if you think it's overkill. It's in `warning-middleware.js`. Figured we could use it for handling more.

Reply:
Doing the warnings there is fine. Just make sure you're returning after you warn. I'll move this convo over to that file.